### PR TITLE
Tm random adjective generator

### PR DESCRIPTION
**Pull request by tova-moskowitz**
First PR. Let's discuss the changes I made to our project.

### CODE REVIEW

File: src/components/Sentence/index.js [hunk 1]
```diff
@@ -16,52 +16,64 @@ const Sentence = React.createClass({
...
+    'their',
     'his',
-    'her'
+    'her',
+    'our'
   ],
-  getWord: function(posp){
-    const index = Math.floor(Math.random() * posp.length);
-    return posp[index];
+  getWord: function(poSp){
```
Comment (tova-moskowitz) [inline src/components/Sentence/index.js]: Will change variable name to poSpArr so it's clear that it's an array

**Comment by devorah** [comment]
Yay! PR #1!! Btw... what's the difference between this branch and the other one (`random_adjective_generator`)?

File: src/components/Sentence/index.js (hunk 1 above)
Comment (devorah) [reply src/components/Sentence/index.js]: um, I was going to ask you what `poSp` was. I think `poSpArr` is even more cryptic! Let's try for a very clear name that doesn't require a special decoder ring to decipher :)

**Comment by tova-moskowitz** [comment]
Yay! 
I created the new one with my initials as a prefix so we can see who owns which branch. I should get rid of the other branch (`random_adjective_generator`)

File: src/components/Sentence/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    'our'
+  ],
+  getWord: function(poSp){
+    const index = Math.floor(Math.random() * poSp.length);
+    return poSp[index];
+  },
+  render: function() {
+    return <div>Oh {this.getWord(this.adjectives)} {this.getWord(this.nounsSubject)}, <br /> how {this.getWord(this.adjectives)} {this.getWord(this.pronouns)}, <br/> when {this.getWord(this.possessives)} {this.getWord(this.nounsObject)} doth  {this.getWord(this.verbs)}</div>;
+  },
+});
```
Comment (devorah) [inline src/components/Sentence/index.js:76]: This is wonderful!

You may experiment with the idea of using template strings, which would help for:
1. readability of your sentence template and 
2. would easily allow us to switch out and use different "modes", i.e. different templates. 

Like this:

``` js
const adjective = this.getWord(this.adjectives);
const noun = this.getWord(this.nouns);
// ...

return (
    `Oh ${adjective} ${noun}, <br />
    how ${adjective}` ${pronoun}, <br /> ...
);
```

**Comment by devorah** [comment]
You can get rid of it if contains changes we no longer need...

File: src/components/Sentence/index.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  ],
+  possessives: [
+    'my',
+    'your',
+    'their',
+    'his',
+    'her',
+    'our'
+  ],
+  getWord: function(poSp){
```
Comment (devorah) [inline src/components/Sentence/index.js]: Let's try for more descriptive argument names that do not require a decoder ring to decipher :) I'm actually not sure what `poSp` means. Can you try for something clearer?

File: src/components/Sentence/index.js (hunk 2 above)
Comment (tova-moskowitz) [reply src/components/Sentence/index.js:76]: Oh cool, that's so clean! When can we get together and work on this??

File: src/components/App/index.js [hunk 4]
```diff
@@ -3,14 +3,24 @@
...
+import Sentence from '../Sentence';
 
-function App() {
-  return (
-    <div>
-      <HelloWorld />
-    </div>
-  );
-}
+// function App() {
```
Comment (devorah) [inline src/components/App/index.js:9]: Let's remove this commented code if we no longer need it.

File: src/components/Sentence/index.js (hunk 2 above)
Comment (devorah) [reply src/components/Sentence/index.js:76]: Tomorrow!

**Comment by tova-moskowitz** [comment]
This is great! Your suggestions are wonderful!

File: src/components/Sentence/index.js (hunk 3 above)
Comment (tova-moskowitz) [reply src/components/Sentence/index.js]: It stands for part of speech, but only after using a decoder ring ;)
I'm afraid of using long function names. What's the accepted protocol for that?

Comment (devorah) [reply src/components/Sentence/index.js]: Don't be afraid of long function or variable names. Clarity/readability/understandability come before brevity or cleverness. In my codebase at work, we have a function named `setDesiredSuspensionRestartDateAndValidateDates`. It's very clear what it does, right? Imagine if instead it said `setSusResDateVal`!

Comment (devorah) [reply src/components/Sentence/index.js]: http://martinfowler.com/bliki/TwoHardThings.html

Comment (devorah) [reply src/components/Sentence/index.js]: http://martinfowler.com/bliki/TwoHardThings.html

File: src/components/App/index.js (hunk 4 above)
Comment (tova-moskowitz) [reply src/components/App/index.js:9]: Decided to keep this, as example of ECMA6 and stateless functions

Comment (devorah) [reply src/components/App/index.js:9]: Ok great!

File: src/components/Sentence/index.js [hunk 5]
```diff
@@ -68,9 +66,9 @@ const Sentence = React.createClass({
     'her',
     'our'
   ],
-  getWord: function(poSp){
-    const index = Math.floor(Math.random() * poSp.length);
-    return poSp[index];
+  getWord: function(wordsArray){
```
Comment (devorah) [inline src/components/Sentence/index.js:69]: Nice choice!

File: src/components/Sentence/index.js (hunk 2 above)
Comment (devorah) [reply src/components/Sentence/index.js:76]: Do you think it's worth doing this refactor in this PR or perhaps in a subsequent one?

Comment (tova-moskowitz) [reply src/components/Sentence/index.js:76]: I was thinking the same thing- it's more of a structural thing and maybewe can deal with it later

**Comment by devorah** [comment]
☀️

File: src/components/Sentence/index.js (hunk 2 above)
Comment (devorah) [reply src/components/Sentence/index.js:76]: Sounds good!

**Pull request merged**